### DISCUSSION TITLE
ESP8266 not connecting to MQTT broker

### DISCUSSION THREAD

**Discussion by biz2000** [arendst/Tasmota#22593] [Support request]
Hi,

I have troubles trying to connect an ESP8266 to my MQTT broker.

Error is rc -2, so network error.

![Capture d'écran 2024-12-06 100450](https://github.com/user-attachments/assets/be3d613b-99f3-45a2-8643-5079598e3dd5)

Here is my information page:

![Capture d'écran 2024-12-06 100346](https://github.com/user-attachments/assets/586300aa-7c29-4f64-b19b-ed5abdd43a92)

I tried with 3 (brand new) different devices.
Initially i had installed Mosquitto on a LXC container in proxmox.
I tried to install EQMX, as it was not working properly with Mosquitto.

In both cases, the connection works well either with MQTT explorer on my pc inside the LAN.
It also works with my smartphone, via a duckdns url.
ESP connection with a public MQTT broker is ok.

Ports are open on my provider's box.
When i ping the ESP from the VM i use (only one VM in my proxmox setup), it works. The HTTP interface works as well.
I have tried to perform a tcpdump on various network interfaces and i did not notice any packet sent from or to the ESP ip address related to MQTT.
i have tried the mqtttimeout increased trick seen on another discussion.

I really don't know what to do next.
Thanks!

**Comment by Jason2866** [comment]
Which device? Such strange issues happens when the esp has a bad power supply.

**Reply by biz2000** [reply]
I have tried with 2 ESP01, one ESP 12S, powered either by my pc, my laptop or a phone charger.
